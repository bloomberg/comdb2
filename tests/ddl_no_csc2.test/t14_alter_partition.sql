DROP TABLE IF EXISTS t14_1
SELECT '- CREATE A TIME PARTITION -' AS caption
CREATE TABLE t14_1 (a INT) PARTITIONED BY TIME PERIOD 'daily' RETENTION 7 start '2023-01-01T00:00:00 UTC'$$
EXEC PROCEDURE sys.cmd.send("partinfo")
SELECT '- INSERT DATA -' AS caption
INSERT INTO t14_1 VALUES(1)
SELECT 'ALTER PARTITIONED BY - THIS SHOULD FAIL' AS caption
ALTER TABLE t14_1 PARTITIONED BY TIME PERIOD 'monthly' RETENTION 6 start '2023-01-01T00:00:00 UTC'$$
SELECT 'ALTER REPARTITIONED BY - ADD A COLUMN, AND INCREASE RETENTION' AS caption
ALTER TABLE t14_1 ADD COLUMN b INT, REPARTITIONED BY TIME PERIOD 'daily' RETENTION 14 start '2023-01-01T00:00:00 UTC'$$
EXEC PROCEDURE sys.cmd.send("partinfo")
SELECT 'INSERT DATA' AS caption
INSERT INTO t14_1 VALUES(2)
INSERT INTO t14_1 VALUES(2, 2)
SELECT * FROM t14_1
SELECT 'ALTER REPARTITIONED BY - DROP A COLUMN, AND REDUCE RETENTION' AS caption
ALTER TABLE t14_1 DROP COLUMN b, REPARTITIONED BY TIME PERIOD 'monthly' RETENTION 6 start '2023-01-01T00:00:00 UTC'$$
SELECT 'INSERT DATA' AS caption
INSERT INTO t14_1 VALUES(3)
INSERT INTO t14_1 VALUES(3, 3)
SELECT * FROM t14_1
EXEC PROCEDURE sys.cmd.send("partinfo")
