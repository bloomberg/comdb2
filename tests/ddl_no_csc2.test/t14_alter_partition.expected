(caption='- CREATE A TIME PARTITION -')
(out='Partition "t14_1":')
(out=' {')
(out='  "NAME"      : "t14_1",')
(out='  "PERIOD"    : "daily",')
(out='  "RETENTION" : 7,')
(out='  "STARTTIME" : "2022-12-31T190000.000 America/New_York",')
(out='  "SHARD0NAME": "<none>",')
(out='  "SOURCE_ID" : "1f9a4021-89ba-41fe-96e5-9e4ff9ba9c0e",')
(out='  "ROLLOUT"   : "TRUNCATE",')
(out='  "TABLES"    :')
(out='  [')
(out='  {')
(out='   "TABLENAME"    : "$0_F79FC69",')
(out='   "LOW"          : 1901-12-13T154552.000 America/New_York,')
(out='   "HIGH"         : 2023-09-23T200000.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$1_7346C562",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$2_2577B4DF",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$3_B604231",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$4_12922A14",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$5_81CAE45E",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$6_9D10E03E",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  }')
(out='  ]')
(out=' }')
(out='')
(caption='- INSERT DATA -')
(rows inserted=1)
(caption='ALTER PARTITIONED BY - THIS SHOULD FAIL')
[ALTER TABLE t14_1 PARTITIONED BY TIME PERIOD 'monthly' RETENTION 6 start '2023-01-01T00:00:00 UTC'] failed with rc -3 Partition already exists
(caption='ALTER REPARTITIONED BY - ADD A COLUMN, AND INCREASE RETENTION')
(out='Partition "t14_1":')
(out=' {')
(out='  "NAME"      : "t14_1",')
(out='  "PERIOD"    : "daily",')
(out='  "RETENTION" : 14,')
(out='  "STARTTIME" : "2022-12-31T190000.000 America/New_York",')
(out='  "SHARD0NAME": "<none>",')
(out='  "SOURCE_ID" : "77f85982-e33c-4313-8a43-0be054d74793",')
(out='  "ROLLOUT"   : "TRUNCATE",')
(out='  "TABLES"    :')
(out='  [')
(out='  {')
(out='   "TABLENAME"    : "$0_F79FC69",')
(out='   "LOW"          : 1901-12-13T154552.000 America/New_York,')
(out='   "HIGH"         : 2023-09-23T200000.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$1_7346C562",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$2_2577B4DF",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$3_B604231",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$4_12922A14",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$5_81CAE45E",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$6_9D10E03E",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$7_57B4DD7A",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$8_874C26A4",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$9_521F4EDC",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$10_F14BAB68",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$11_E4633C3B",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$12_E031DE34",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$13_A9D5DC1",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  }')
(out='  ]')
(out=' }')
(out='')
(caption='INSERT DATA')
[INSERT INTO t14_1 VALUES(2)] failed with rc -3 table t14_1 has 2 columns but 1 values were supplied
(rows inserted=1)
(a=1, b=NULL)
(a=2, b=2)
(caption='ALTER REPARTITIONED BY - DROP A COLUMN, AND REDUCE RETENTION')
(caption='INSERT DATA')
(rows inserted=1)
[INSERT INTO t14_1 VALUES(3, 3)] failed with rc -3 table t14_1 has 1 columns but 2 values were supplied
(a=3)
(a=1)
(a=2)
(out='Partition "t14_1":')
(out=' {')
(out='  "NAME"      : "t14_1",')
(out='  "PERIOD"    : "monthly",')
(out='  "RETENTION" : 6,')
(out='  "STARTTIME" : "2022-12-31T190000.000 America/New_York",')
(out='  "SHARD0NAME": "<none>",')
(out='  "SOURCE_ID" : "c65b9157-63ab-450e-afb8-8ffc343a3b7a",')
(out='  "ROLLOUT"   : "TRUNCATE",')
(out='  "TABLES"    :')
(out='  [')
(out='  {')
(out='   "TABLENAME"    : "$0_F79FC69",')
(out='   "LOW"          : 1901-12-13T154552.000 America/New_York,')
(out='   "HIGH"         : 2023-09-30T200000.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$1_7346C562",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$2_2577B4DF",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$3_B604231",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$4_12922A14",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  },')
(out='  {')
(out='   "TABLENAME"    : "$5_81CAE45E",')
(out='   "LOW"          : 2038-01-18T221407.000 America/New_York,')
(out='   "HIGH"         : 2038-01-18T221407.000 America/New_York')
(out='  }')
(out='  ]')
(out=' }')
(out='')
